compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt

```diff
@@ -46,29 +46,60 @@ private class DisplayLinkConditions(
         }
 
     /**
-     * Indicates that scene is invalidated and next display link callback will draw
+     * When CADisplayLink invokes a callback, this value is checked to decide if draw needs to be dispatched
      */
-    var needsRedrawOnNextVsync: Boolean = false
+    val needsRedrawOnNextVsync: Boolean
+        get() = scheduledRedrawsCount > 0
+
+    /**
+     * Indicates that application is running foreground now
+     */
+    var isApplicationActive: Boolean = false
         set(value) {
             field = value
 
             update()
         }
 
     /**
-     * Indicates that application is running foreground now
+     * Number of subsequent vsync that will issue a draw
      */
-    var isApplicationActive: Boolean = false
+    private var scheduledRedrawsCount = 0
         set(value) {
             field = value
 
             update()
         }
 
+    /**
+     * Notify that vsync has issued a draw dispatch
+     */
+    fun onDrawStart() {
```

Comment:
Shouldn't we call it `callbackStarted` and call it from the display link callback?

Because it seems that it is coupled with display link logic.

Besides, we can force redraw outside display link, and we shouldn't decrease `scheduledRedrawsCount` in that case, otherwise we will have 60hz.

If we will do it this way, we need to:
1. return `needsRedrawOnNextVsync`
2. rename `scheduledRedrawsCount` to `scheduledCallbackCount`
3. do this change:
```
val isUnpaused = isApplicationActive && (needsToBeProactive || needsRedrawOnNextVsync || scheduledCallbackCount > 0)
```

Reply:
> Shouldn't we call it callbackStarted and call it from the display link callback?
I think both reflect the intent well.

> Because it seems that it is coupled with display link logic.
As it should be, the class is called "DisplayLinkConditions" for that specific reason.

> Besides, we can force redraw outside display link, and we shouldn't decrease scheduledRedrawsCount in that case, otherwise we will have 60hz.
We won't. `onDrawStart` is called inside `handleDisplayLinkTick` which is a displayLink callback. 

All this logic is related specifically to interaction with CADisplayLink. Sync redraws (which only now happen on first appearance) don't interact with it.

> return needsRedrawOnNextVsync
It didn't go anywhere, it's just redundant to denormalise the state, which can be derived from `scheduledRedrawsCount`. And the variable itself simply indicates how many subsequent vsyncs will issue the draw, as per doc, as opposed to _blank_ vsyncs (during touch tracking), when vsync happens, but no draws will be issued this frame.

Reply:
Have a look. Does https://github.com/JetBrains/compose-multiplatform-core/pull/797/commits/1f58a72fc7696d342d7102810c324d5644dfc339 make it easier to follow?

Reply:
> We won't. onDrawStart is called inside handleDisplayLinkTick which is a displayLink callback.

My mistake, didn't notice that because of the function name :).

> Have a look. Does https://github.com/JetBrains/compose-multiplatform-core/commit/1f58a72fc7696d342d7102810c324d5644dfc339 make it easier to follow?

Yes, this variant is good